ale_linters/psy/psyko.vim

```diff
@@ -0,0 +1,85 @@
+" Description: Integration with ALE to run psyko as a linter
+" Version: Compatible with ASTERIOS Developer K19.2
+
+call ale#Set('psy_psyko_executable', 'psyko')
+call ale#Set('psy_psyko_product', 'ksim')
+call ale#Set('psy_psyko_kernel_dir', '')
+call ale#Set('psy_psyko_options', '--color no')
+
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+let s:psyko_error_regex =
+            \ '\v\w+\>(fatal|error|warning)\>' .
+            \ '\s+(\w+):\s+((.*):(\d+):(\d+):\s+)?(.*)$'
+
+let s:psyko_log_msglevel = {
+                    \ 'fatal': 'E',
+                    \ 'error': 'E',
+                    \ 'warning': 'W',
+                    \ '': 'E'
+                    \}
+
+let s:psyko_root_error_msg_regex = '\v\(faulty command below\):'
+
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+function! s:GetExecutable(buffer) abort
+    let l:exec = ale#Var(a:buffer, 'psy_psyko_executable')
+
+    " TODO: handle gracefuly error cases:
+    "
+    "   - incorrect psyko version
+    "   - psy_psyko_kernel_dir doesn't exist
+
+    return l:exec
+endfunction
+
+
+function! s:GetCommand(buffer) abort
+    " prepare a temp file for the output db
+    let l:tmpoutputdb = tempname() . '.ks'
+    call ale#command#ManageFile(a:buffer, l:tmpoutputdb)
+
+    return '%e '
+    \   . ale#Var(a:buffer, 'psy_psyko_options')
+    \   . ' --product ' . ale#Var(a:buffer, 'psy_psyko_product')
+    \   . ' --kernel-dir ' . ale#Var(a:buffer, 'psy_psyko_kernel_dir')
+    \   . ' module --gendb'
+    \   . ' --output ' . l:tmpoutputdb
+    \   . ' -- ' . ale#Escape(fnamemodify(bufname(a:buffer), ':p'))
```

Comment:
Very often, psy_pp_flags need to be provided in a psymodule configuration file.
This is necessary to add include directories for instance.

Maybe a global ale configuration variable that points on a configuration file is enough for now... I do not see any simple solution to configure this by project.

Reply:
I thought I would to take a _kbuildgen_ (-ish) configuration file for project configuration, in a not too distant future. This may even allow for smarter linting messages, if we can push up to `psyko partition` for instance.

In the meantime, ALE allows for global or buffer variables (e.g. `b:ale_psy_psyko_kernel_dir` takes precedednce over `g:ale_psy_psyko_kernel_dir`), which is better than nothing. So I propose I add an optional `psy_psyko_conf_files` variable to list additional JSON configuration files to be passed to `psyko module`. Deal?